kernel/src/main/scala/cats/kernel/std/double.scala

```diff
@@ -0,0 +1,26 @@
+package cats.kernel
+package std
+
+import java.lang.Math
+
+trait DoubleInstances {
+  implicit val doubleOrder: Order[Double] = new DoubleOrder
+}
+
+class DoubleOrder extends Order[Double] {
```

Comment:
strictly, is this a PartialOrder due to `NaN`? Should we maybe have that and some `unsafeToOrder` method for people to opt out?